CDCgov/prime-reportstream
# Platform/jessica/12816 original message passthrough

This PR implements the [design of original message passthrough](https://github.com/CDCgov/prime-reportstream/blob/master/prime-router/docs/design/proposals/0024-original-message-passthrough.md), which allows the original message sent by the sender to be what the receiver receives, instead of receiving a translated file.  There are numerous shortcomings (tech debt) in this ticket which will be picked up immediately in #13232 but the core functionality is tested and working. #13232 includes a pretty exhaustive list of what we would like to update.

Test Steps:
1. Send a message, either FHIR or HL7v2, under the` elr-elims` topic. Try this with and without setting `payloadName` in the request header.
3. Verify SFTP folder in build directory contains the expected message, that the message is IDENTICAL 100% to the original message you sent, and that the file name includes payloadName if you provided that in the submission

## Changes
- `Topic` enum updated with `isSendOriginal` property that modifies behavior of the translate function
- Update translate function to consider `isSendOriginal` topic property for business logic
- Updated "database integration test" 

## Checklist

### Testing

## Linked Issues
- #issue 12816
